File: ios/ios/Core/Models/GeneratedSong.swift
import SwiftUI

struct GeneratedSong: Identifiable, Hashable {
    let id: String
    let title: String
    let subtitle: String
    let color: Color
    let participantCount: Int
    let generatedAt: Date?
    let durationSec: Int?
    let mood: String?
    let myLyric: String?
    let audioURL: String?
    let chainId: String?
    let isLiked: Bool
}

File: backend/internal/domain/entity/generated_song.go
package entity

import (
	"time"

	"github.com/google/uuid"

	"hackathon/internal/domain/vo"
)

// GeneratedSong は Lyria で生成された楽曲。
type GeneratedSong struct {
	ID          string
	ChainID     string
	Title       *string
	AudioURL    *string
	DurationSec *int
	Mood        *string
	Genre       *string
	Status      vo.GeneratedSongStatus
	GeneratedAt *time.Time
}

// NewGeneratedSong は processing 状態の GeneratedSong を生成する。
func NewGeneratedSong(chainID string) GeneratedSong {
	return GeneratedSong{
		ID:      uuid.NewString(),
		ChainID: chainID,
		Status:  vo.GeneratedSongStatusProcessing,
	}
}
